src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java

```diff
@@ -1,16 +1,40 @@
 package uk.gov.companieshouse.overseasentitiesapi.validation;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
 import uk.gov.companieshouse.overseasentitiesapi.model.dto.EntityDto;
 import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
 import uk.gov.companieshouse.overseasentitiesapi.validation.utils.StringValidators;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators;
 import uk.gov.companieshouse.service.rest.err.Errors;
 
 @Component
 public class EntityDtoValidator {
 
+    private final AddressDtoValidator addressDtoValidator;
+
+    @Autowired
+    public EntityDtoValidator(AddressDtoValidator addressDtoValidator) {
+        this.addressDtoValidator = addressDtoValidator;
+    }
+
     public Errors validate(EntityDto entityDto, Errors errs, String loggingContext) {
         validateName(entityDto.getName(), errs, loggingContext);
+        validateIncorporationCountry(entityDto.getIncorporationCountry(), errs, loggingContext);
+        validatePrincipalAddress(entityDto.getPrincipalAddress(), errs, loggingContext);
+        boolean sameFlagValid = validateServiceAddressSameAsPrincipalAddress(entityDto.getServiceAddressSameAsPrincipalAddress(), errs, loggingContext);
```

Comment:
I'm not sure sure what 'sameFlagValid' means or represents. Can it be better named?

Reply:
Tried to name it as best I can.